src/kanata/mod.rs

```diff
@@ -1349,10 +1349,15 @@ impl Kanata {
         self.cur_keys.extend(self.layout.bm().keycodes());
         self.overrides
             .override_keys(&mut self.cur_keys, &mut self.override_states);
-        let current_layer = self.layout.bm().current_layer();
-        if current_layer % 2 == 1 {
-            // Prioritize checking the active layer in case a layer-while-held is active.
-            if let Some(outputs_for_key) = self.key_outputs[current_layer].get(&event.code) {
+
+        // Prioritize checking the active layer in case a layer-while-held is active.
+        let active_held_layers = self.layout.bm().active_held_layers();
+        let mut held_layer_active = false;
+        for layer in active_held_layers {
+            if !held_layer_active {
```

Comment:
This conditional doesn't need to be here, can always just set to `true` to avoid a branch (though it may be compiled away anyway, but best not to rely on chance).

Reply:
I wasn't sure what's faster, since reading memory is faster than writing, but on the other hand there's a branch cost.

Reply:
I'm not convinced what's faster, I would have to see a benchmark to believe, but I'm too lazy to do it. Either way, removing the conditional produces shorter code, let's remove it.

Reply:
041e09e23eaefd263c26173df46098d37b897030

Reply:
My understanding of this topic is that a non-atomic-read+branch will always be slower than a single non-atomic write. Branch prediction exists to help the speculative execution, but may as well avoid relying on it where we can!